kmod/patch/kpatch.lds.S

```diff
@@ -9,6 +9,29 @@ __kpatch_checksum = ADDR(.kpatch.checksum);
 
 SECTIONS
 {
+  .text.unlikely.livepatch : {
+    *(.text.unlikely.*)
+  }
+
+  .text.livepatch : {
+    *(.text.*)
```

Comment:
FWIW, this pattern seems compatible with scripts/module-common.lds and arch/powerpc/kernel/module.lds (no x86_64 counterpart). 

I did notice that arch/x86/kernel/vmlinux.lds.S does use the linker script to setup `__indirect_thunk_{start,end}` relocations like so:
```
SECTIONS
{
...
        /* Text and read-only data */
        .text :  AT(ADDR(.text) - LOAD_OFFSET) {
...
                __indirect_thunk_start = .;
                *(.text.__x86.indirect_thunk)
                __indirect_thunk_end = .;
```
and I wonder if a pattern like this ever made it into one of the module.lds files if we're going to experience some weird failure modes.  I don't know much about module linker files, so this only speculation.  For another example, arch/arm64/kernel/module.lds includes this:
```
SECTIONS {
...
	.text.ftrace_trampoline (NOLOAD) : { BYTE(0) }
}
```
so maybe `*(.text.*)` is too inclusive and we should try to limit it to the functions that kpatch-build detects have changed?

Reply:
> so maybe *(.text.*) is too inclusive and we should try to limit it to the functions that kpatch-build detects have changed?

You mean, generate kpatch.lds on the fly, with the patterns corresponding to changed functions like

Reply:
Yeah that was the suggestion, but maybe I'm being too paranoid, I don't know.

@jpoimboe any thoughts on this patch module linker script?

Reply:
As @euspectre said, merging sections is not an option for FGKASLR, so wouldn't this make kpatch incompatible with FGKASLR?  Maybe the proper fix is indeed to use vmalloc.